src/frontend/src/handler/alter_table_column.rs

```diff
@@ -193,6 +187,24 @@ pub async fn handle_alter_table_column(
                 bail_not_implemented!(issue = 6903, "drop column cascade");
             }
 
+            // Check if the column to drop is referenced by any generated columns.
+            for column in original_catalog.columns() {
+                if let Some(expr) = column.generated_expr() {
+                    let expr = ExprImpl::from_expr_proto(expr)?;
+                    let refs = expr.collect_input_refs(original_catalog.columns().len());
+                    for idx in refs.ones() {
+                        let refed_column = &original_catalog.columns()[idx];
+                        if refed_column.name() == column_name.real_value() {
+                            Err(ErrorCode::PermissionDenied(format!(
```

Comment:
Permission denied does not sound to be an appropriate category. 🤣

Reply:
I don't like it either. But we are using it everywhere https://github.com/singularity-data/risingwave/blob/739d0059f3b7a76975d503821549fd808fef8af9/src/meta/src/manager/catalog/mod.rs#L592